lib/srv/discovery/fetchers/db/aws.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package db
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/discovery/common"
+)
+
+// MakeAWSFetchers creates fetchers for AWS-hosted databases.
+func MakeAWSFetchers(clients cloud.Clients, matchers []services.AWSMatcher) (result []common.Fetcher, err error) {
+	type makeFetcherFunc func(cloud.Clients, string, types.Labels) (common.Fetcher, error)
+	makeFetcherFuncs := map[string][]makeFetcherFunc{
+		services.AWSMatcherRDS:         {makeRDSInstanceFetcher, makeRDSAuroraFetcher},
+		services.AWSMatcherRDSProxy:    {makeRDSProxyFetcher},
+		services.AWSMatcherRedshift:    {makeRedshiftFetcher},
+		services.AWSMatcherElastiCache: {makeElastiCacheFetcher},
+		services.AWSMatcherMemoryDB:    {makeMemoryDBFetcher},
+	}
+
+	for _, matcher := range matchers {
+		for _, matcherType := range matcher.Types {
+			makeFetchers, found := makeFetcherFuncs[matcherType]
+			if !found {
+				return nil, trace.BadParameter("unknown matcher type %q", matcherType)
+			}
```

Comment:
added this check for unknown matcher similar how `MakeAzureFetchers` does it right now.